### ISSUE TITLE
[Flaky Test]: TestLongRunningAgentForLeaks/TestHandleLeak – Condition never satisfied

### ISSUE THREAD

**Issue by rdner** [elastic/elastic-agent#4447]
### Failing test case

TestLongRunningAgentForLeaks/TestHandleLeak

### Error message

Condition never satisfied

### Build

https://buildkite.com/elastic/elastic-agent/builds/7876#018e5c3a-00ae-4972-b890-37cad38f2a96

### OS

Windows

### Stacktrace and notes

```markdown
agent_long_running_leak_test.go:232: component state: Healthy: communicating with pid '3216'
    agent_long_running_leak_test.go:223: unit ID: winlog-fleet-default-output
    agent_long_running_leak_test.go:223: unit ID: winlog-fleet-default-output-winlog-system-e3f438c4-571c-44b0-a8d5-25e4ea556eef
    agent_long_running_leak_test.go:232: component state: Healthy: communicating with pid '2344'
    agent_long_running_leak_test.go:210: 
        	Error Trace:	C:/Users/windows/agent/testing/integration/agent_long_running_leak_test.go:210
        	            				C:/Users/windows/agent/testing/integration/agent_long_running_leak_test.go:142
        	Error:      	Condition never satisfied
        	Test:       	TestLongRunningAgentForLeaks/TestHandleLeak
        	Messages:   	install never became healthy
--- FAIL: TestLongRunningAgentForLeaks/TestHandleLeak (180.00s)
```

**Comment by rdner** [comment]
It failed in the daily build too, perhaps it's a real leak then https://buildkite.com/elastic/elastic-agent/builds/7884#018e5e50-9390-41d6-855d-2979484718c7

**Comment by rdner** [comment]
I skipped the tests for now to avoid more failures, needs to be investigated ASAP
https://github.com/elastic/elastic-agent/pull/4452

**Comment by cmacknz** [comment]
Are we sure this is a bug in this test, and it is not that this test is finding a real bug in the underlying inputs it is testing?

**Comment by rdner** [comment]
@cmacknz it can be a real failure, we need to properly investigate of course. The most concerning fact is that it started failing without any connection to a new change in the code.

It first started failing after https://github.com/elastic/elastic-agent/issues/4447 was merged. Initially, I thought this change introduced something that fails the test, despite the PR being green.

However, then I opened a new PR to revert this change https://github.com/elastic/elastic-agent/pull/4451 and in that PR the leak tests also failed which proves that the failure is not related to the change but just came out of nowhere.
